R/settings_utils.R

```diff
@@ -69,24 +69,24 @@ find_config <- function(filename) {
     dirname(filename)
   }
 
-  ## check for a file in the current directory
-  linter_config <- file.path(path, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file higher directories
-  linter_config <- find_config2(path)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
-  }
-
-  ## next check for a file in the user directory
-  # cf: rstudio@bc9b6a5 SessionRSConnect.R#L32
-  home_dir <- Sys.getenv("HOME", unset = "~")
-  linter_config <- file.path(home_dir, linter_file)
-  if (isTRUE(file.exists(linter_config))) {
-    return(linter_config)
+  # List possible .lintr file locations
```

Comment:
I think we can be a bit more concise w the comments on the list and move the "meat" to the top where it's a little less distracting.

the first comment can add "NB: this vector specifies a priority order for where to find the configs, i.e. the first location where a config exists is chosen and configs which may exist in subsequent directories are ignored"

then in the item comments we can e.g. change "check for a file in the current directory" to just "current directory" etc

Reply:
Thanks @MichaelChirico , updated the code per the comment above